base/range.jl

```diff
@@ -822,12 +837,27 @@ end
 
 getindex(r::AbstractRange, ::Colon) = copy(r)
 
-function getindex(r::AbstractUnitRange, s::AbstractUnitRange{<:Integer})
+function getindex(r::AbstractUnitRange, s::AbstractUnitRange{T}) where {T<:Integer}
     @_inline_meta
     @boundscheck checkbounds(r, s)
-    f = first(r)
-    st = oftype(f, f + first(s)-1)
-    range(st, length=length(s))
+
+    if T === Bool
+        if length(s) == 0
+            return r
+        elseif length(s) == 1
+            if first(s)
+                return r
+            else
+                return range(r[1], length=0)
```

Comment:
```suggestion
                return range(first(r), length=0)
```

In case someone makes an OffsetUnitRange?